src/App.tsx

```diff
@@ -1,27 +1,210 @@
-import React from 'react';
-import './App.scss';
-
-interface Props {
-  onClick: () => void;
-}
-
-export const Provider: React.FC<Props> = React.memo(
-  ({ onClick, children }) => (
-    <button
-      type="button"
-      onClick={onClick}
-    >
-      {children}
-    </button>
-  ),
-);
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { Todo } from './types/Todo';
+import { FilterBy, ErrorMessage } from './types/Constants';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+import { RequestTodo } from './types/RequestTodo';
+import { client } from './utils/fetchClient';
+import { Header } from './Components/Header';
+import { TodoList } from './Components/TodoList';
+import { Footer } from './Components/Footer';
+import { ErrorModal } from './Components/ErrorModal';
+
+const USER_ID = 10552;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [querySearch, setQuerySearch] = useState('');
+  const [filterBy, setFilterBy] = useState<FilterBy>(FilterBy.ALL);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterBy) {
+        case FilterBy.ACTIVE:
+          return !todo.completed;
+        case FilterBy.COMPLETED:
+          return todo.completed;
+        case FilterBy.ALL:
+          return true;
+        default:
+          return todo;
```

Comment:
Switch case should return always the same type of data. Why all cases return booleans and default returns an object?